### ISSUE TITLE
auto-update of devon.bat not working

### ISSUE THREAD

**Issue by hohwille** [devonfw/devon-ide#235]
```
C:\projects\cool-project>devon -v
C:\projects\cool-project
3.0.0-beta18

C:\projects\cool-project>scripts\devon -v
3.0.0-beta21

C:\projects\cool-project>scripts\devon
devon-ide environment variables have been set for C:\projects\cool-project in workspace main

C:\projects\cool-project>devon -v
C:\projects\cool-project
3.0.0-beta18
```

Expectation would be that the `devon.bat` would update itself in `~/scripts` so the last output should have been `3.0.0-beta21`.

**Comment by hohwille** [comment]
Running `setup` again:
```
C:\projects\cool-project>setup
Setting up your devon-ide in C:\projects\cool-project
"Updating devon.bat if newer"
0 File(s) copied
```

https://github.com/devonfw/devon-ide/blob/1b0a4492fc8982ce74d44a7868564c9c1b0a52fd/scripts/src/main/resources/scripts/devon.bat#L21

Seems as if the XCOPY documentation is wrong or I am missing something.

**Comment by hohwille** [comment]
https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/xcopy

> 

/d [:MM-DD-YYYY] | Copies source files changed on or after the specified date only. If you do not include a MM-DD-YYYY value, xcopy copies all Source files that are newer than existing Destination files. This command-line option allows you to update files that have changed.
-- | --

**Comment by hohwille** [comment]
I could determine the installed version like this:
```
for /F "tokens=*" %%g in ('%USERPROFILE%\scripts\devon.bat -v') do (SET "CURRENT_DEVON_VERSION=%%g")
```
And also get the current version from the script (`$[devon_ide_version]`).
However, I am not keen on implementing the version compare algorithm also in CMD.
That was horrible enough to do in Bash.
However, it also seems kind of odd to expose a CLI command for the version compare so I could compare it with existing bash functionality from the `devon.bat` script.
Instead, I think it is better to remove this from `devon.bat` at all and add the logic to `ide` comandlet for `setup`.

**Comment by hohwille** [comment]
Fixed with PR #241
